oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -511,25 +498,69 @@ int mca_spml_ucx_deregister(sshmem_mkey_t *mkeys)
         ucp_mem_unmap(mca_spml_ucx.ucp_context, ucx_mkey->mem_h);
     }
     ucp_rkey_destroy(ucx_mkey->rkey);
+    ucx_mkey->rkey = NULL;
 
     if (0 < mkeys[0].len) {
         ucp_rkey_buffer_release(mkeys[0].u.data);
     }
 
+    free(mkeys);
+
     return OSHMEM_SUCCESS;
 }
 
+static inline void _ctx_add(mca_spml_ucx_ctx_array_t *array, mca_spml_ucx_ctx_t *ctx)
+{
+    int i;
+
+    if (array->ctxs_count < array->ctxs_num) {
+        array->ctxs[array->ctxs_count] = ctx;
+    }
+    else {
+        array->ctxs = realloc(array->ctxs, (array->ctxs_num + 8) * sizeof(mca_spml_ucx_ctx_t *));
```

Comment:
better use macro definition for 8 instead of number

Reply:
done in latest commit